test/e2e/storage/testsuites/snapshottable.go

```diff
@@ -79,167 +82,313 @@ func (s *snapshottableTestSuite) SkipRedundantSuite(driver TestDriver, pattern t
+			err = WaitForGVRDeletion(l.dc, SnapshotContentGVR, snapshotContentName, framework.Poll, framework.SnapshotDeleteTimeout)
+			framework.ExpectNoError(err)
+		} else if sct.DeletionPolicy == "Retain" {
+			ginkgo.By("checking the SnapshotContent has not been deleted")
+			_, err = l.dc.Resource(SnapshotContentGVR).Get(context.TODO(), snapshotContentName, metav1.GetOptions{})
+			framework.ExpectNoError(err)
+
+			// prevent physical snapshotContent leaks
+			defer func() {
+				ginkgo.By("updating the the SnapshotClass to Delete Deletion policy")
+				l.vsc.Object["deletionPolicy"] = "Delete"
+				l.vsc, err = l.dc.Resource(SnapshotClassGVR).Update(context.TODO(), l.vsc, metav1.UpdateOptions{})
+				framework.ExpectNoError(err)
+
+				ginkgo.By("manually deleting the SnapshotContent")
+				err = l.dc.Resource(SnapshotContentGVR).Delete(context.TODO(), l.vscontent.GetName(), metav1.DeleteOptions{})
+				if err != nil && !apierrors.IsNotFound(err) {
+					framework.Failf("Error deleting snapshot content %q. Error: %v", l.vscontent.GetName(), err)
+				}
+
+				ginkgo.By("checking the SnapshotContent has been deleted")
+				err = WaitForGVRDeletion(l.dc, SnapshotContentGVR, snapshotContentName, framework.Poll, framework.SnapshotDeleteTimeout)
+				framework.ExpectNoError(err)
+			}()
+		} else {
+			framework.Failf("Invalid test config. DeletionPolicy should be either Delete or Retain. DeletionPolicy: %v", sct.DeletionPolicy)
+		}
+	}
 
-	ginkgo.It("should create snapshot with defaults [Feature:VolumeSnapshotDataSource]", func() {
-		cs := f.ClientSet
-		dc := f.DynamicClient
+	init := func() {
+		l = local{}
+		l.cs = f.ClientSet
+		l.dc = f.DynamicClient
 
 		// Now do the more expensive test initialization.
 		config, driverCleanup := driver.PrepareTest(f)
-		defer func() {
-			err := tryFunc(driverCleanup)
-			framework.ExpectNoError(err, "while cleaning up driver")
-		}()
-
-		vsc := sDriver.GetSnapshotClass(config)
-		class := dDriver.GetDynamicProvisionStorageClass(config, "")
-		if class == nil {
-			e2eskipper.Skipf("Driver %q does not define Dynamic Provision StorageClass - skipping", driver.GetDriverInfo().Name)
+		l.config = config
+		l.driverCleanup = driverCleanup
+
+		l.vsc = sDriver.GetSnapshotClass(config)
+		l.sc = dDriver.GetDynamicProvisionStorageClass(config, "")
+		if l.sc == nil {
+			framework.Failf("This driver should support dynamic provisioning")
 		}
 		testVolumeSizeRange := s.GetTestSuiteInfo().SupportedSizeRange
 		driverVolumeSizeRange := dDriver.GetDriverInfo().SupportedSizeRange
 		claimSize, err := getSizeRangesIntersection(testVolumeSizeRange, driverVolumeSizeRange)
 		framework.ExpectNoError(err, "determine intersection of test size range %+v and driver size range %+v", testVolumeSizeRange, driverVolumeSizeRange)
-		pvc := e2epv.MakePersistentVolumeClaim(e2epv.PersistentVolumeClaimConfig{
+		l.pvc = e2epv.MakePersistentVolumeClaim(e2epv.PersistentVolumeClaimConfig{
```

Comment:
We can follow up on this separately, but there is a `CreateVolumeResource()` function that handles creating the StorageClass, PVC and Pod

Reply:
Ok sounds good.